CAIPs/caip-341.md

```diff
@@ -0,0 +1,119 @@
+
+```typescript
+interface WalletData {
+  // Required properties
+  uuid: string;
+  name: string;
+  icon: string;
+  rdns: string;
+  // Optional properties
+  target?: {
+    type: "caip341",
+    value: <extension_id>
+  }
+  scopes?: Caip217AuthorizationScopes;
+}
+```
+
+### Usage
+
+The `extensionId` target type is used to specify the extension ID of a browser extension wallet. This allows the dapp to establish a connection with the wallet using the `externally_connectable` API. The `externally_connectable` API documentation can be found [here](https://developer.chrome.com/docs/extensions/reference/manifest/externally-connectable).
+
+```ts
+const walletData = {
+  uuid: "350670db-19fa-4704-a166-e52e178b59d2",
+  name: "Example Wallet",
+  icon: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==",
+  rdns: "com.example.wallet",
+  target: {
+    type: "caip341",
+    value: "abcdefghijklmnopqrstuvwxyz"
+  },
+  scopes: {
+    "eip155:1": {
+      methods: ["eth_signTransaction", "eth_sendTransaction"],
+      notifications: ["accountsChanged", "chainChanged"]
+    }
+  }
+}
+```
+
+### Establishing Connection
+
+When the target type is `extensionId`, the dapp MUST use the `extensionId` to establish a connection with the wallet using the `externally_connectable` browser API. All subsequent communication with the wallet SHOULD be conducted over the `externally_connectable` API using `runtime.connect()` and `runtime.sendMessage()`.
+
+Example of establishing a connection and sending a message:
+
+```ts
+const port = chrome.runtime.connect(walletData.target.value);
+
+port.onMessage.addListener((message) => {
+  // Handle incoming messages
+});
+
+port.postMessage({
+  id: 1,
+  jsonrpc: "2.0",
+  method: "wallet_createSession",
+  params: {
+    // ... session parameters ...
+  }
+});
+```
+
+## Rationale
+
+By defining the `extensionId` target type, we provide a standardized way for dapps to connect with browser extension wallets. This reduces complexity and enhances interoperability across different blockchain ecosystems. The use of the `externally_connectable` API ensures secure and efficient communication between the dapp and the wallet.
+
+## Backwards Compatibility
+
+This CAIP is fully compatible with existing standards and does not introduce any breaking changes. It extends the `walletData` interface to include the `target` field, which is optional and does not affect existing implementations.
+
+## Links
+
+- [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) - Multi Injected Provider Discovery
+- [CAIP-294](https://chainagnostic.org/CAIPs/caip-294) - Browser Wallet Messaging for Extensions
+- [externally_connectable API documentation](https://developer.chrome.com/docs/extensions/reference/manifest/externally-connectable)
```

Comment:
```suggestion

- [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) - Multi Injected Provider Discovery
- [CAIP-294](https://chainagnostic.org/CAIPs/caip-294) - Browser Wallet Messaging for Extensions
- [externally_connectable API documentation](https://developer.chrome.com/docs/extensions/reference/manifest/externally-connectable)
- [Wallet Standard][https://github.com/anza-xyz/wallet-standard]
```

Reply:
resolved in [5b0b6f4](https://github.com/ChainAgnostic/CAIPs/pull/341/commits/5b0b6f465b810ce42d316705db7f1e52302aa7cf)